### ISSUE TITLE
Unable to use legend_toggle

### ISSUE THREAD

**Issue by maccyber** [rakannimer/react-google-charts#196]
Legend toggle gives me the following error when I toggle it.

![g-charts-toggle](https://user-images.githubusercontent.com/701351/40204317-67a181ae-5a28-11e8-961f-7b59346f4e56.png)

Example code in next.js

```js
import { Chart } from 'react-google-charts'

export default () => (
  <Chart
    chartType="ScatterChart"
    data={[['Age', 'Weight'], [8, 12], [4, 5.5]]}
    options={{}}
    graph_id="ScatterChart1"
    width="100%"
    height="400px"
    legend_toggle
  />
)
```

**Comment by igalarza009** [comment]
I had the same problem, and after removing the `legend_toggle` property from the chart, it suddenly started working.

So try this:

    import { Chart } from 'react-google-charts'

    export default () => (
        <Chart
              chartType="ScatterChart"
              data={[['Age', 'Weight'], [8, 12], [4, 5.5]]}
              options={{}}
              graph_id="ScatterChart1"
              width="100%"
              height="400px"
      />
    )

**Comment by maccyber** [comment]
@igalarza009 thanks, but removing `legend_toggle` also removes the ability to toggle.

**Comment by igalarza009** [comment]
@maccyber I actually thought the same before answering the question, but after looking at my code, I realised I had removed it and that the toggle is still working.

However, taking a better look, I also made some changes in the chartEvents, for solving another problem, and it might be the key of why the toggle function still works on my chart even if I don't have the `legend_toggle` specified.

You could try adding some chart event on the `select` property, something like this:

    import React from 'react';
    import {Chart} from 'react-google-charts';

    class MyChart extends React.Component{
       constructor(props) {
	   super(props);
	   this.chartEvents = [
		 {
		     eventName: 'select',
		     callback(Chart) {
			console.log('Selected ', Chart.chart.getSelection());
		     },
	         },
	      ]
	   }

       render(){
           return(
               <Chart
                    chartType="ScatterChart"
                    data={[['Age', 'Weight'], [8, 12], [4, 5.5]]}
                    options={{}}
                    graph_id="ScatterChart1"
                    width="100%"
                   height="400px"
                   chartEvents={this.chartEvents}
               />
           )
       }

    export default () => (
          <MyChart />
     )

I don't know if this could be helpful, but I know had the same exact problem and somehow it got solved. I haven't manually implemented the toggle function neither, The only different thing I have is that I added a chartEvent for when the Chart is `ready`, to solve a completely different problem.

**Comment by maccyber** [comment]
I changed [this line](https://github.com/rakannimer/react-google-charts/blob/master/src/index.js#L364)

From:
```js
let colors = [];
```
To:

```js
let colors = this.props.options && this.props.options.colors ? this.props.options.colors : DEFAULT_CHART_COLORS;
```

That made it work as expected for me. Did not test it with custom colors though

**Comment by rakannimer** [comment]
Hey !

Thanks for reporting the issue.

This should be fixed in latest version (>=2.0).

Feel free to re-open if it re-appears.
